### PR TITLE
feat: Allow users to create dWallets on Pera (DKG)

### PR DESCRIPTION
**Pull request by yaelAbergel**

### CODE REVIEW

File: .cargo/config.toml
Comment (ItayLevyOfficial) [inline .cargo/config.toml:1]: need to undo this change

File: crates/pera-core/src/execution_cache/cache_types.rs [hunk 1]
```diff
@@ -113,12 +113,6 @@ impl<I: Iterator> AssertOrdered<I> {
     }
 }
 
-impl<I: IntoIterator> From<I> for AssertOrdered<I::IntoIter> {
```
Comment (ItayLevyOfficial) [inline crates/pera-core/src/execution_cache/cache_types.rs:116]: This is needed because this implementation conflicts with another implementation from the cryptography-private repository

File: crates/telemetry-subscribers/Cargo.toml [hunk 2]
```diff
@@ -26,7 +26,7 @@ tokio = { workspace = true, features = ["full"] }
 futures.workspace = true
 clap.workspace = true
 bytes.workspace = true
-bytes-varint = { version = "1" }
+bytes-varint = { version = "=1.0.3" }
```
Comment (ItayLevyOfficial) [inline crates/telemetry-subscribers/Cargo.toml:29]: This is needed to use this exact version, as the latest version breaks the project compilation atm

File: sdk/typescript/tsconfig.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
 		"module": "CommonJS",
 		"outDir": "dist/cjs",
 		"rootDir": "src",
-		"isolatedModules": true
+		"isolatedModules": true,
+		"sourceMap": true
```
Comment (ItayLevyOfficial) [inline sdk/typescript/tsconfig.json:9]: Needed for debugging TS code in the IDE

File: crates/telemetry-subscribers/Cargo.toml (hunk 2 above)
Comment (erik-3milabs) [reply crates/telemetry-subscribers/Cargo.toml:29]: 1. Consider turning this into an issue: this will need to be resolved one way or another, even if this PR is not the place for it.
2. Instead of _fixing_ the version at exactly `=1.0.3`, consider using something like `<1.1.0` instead. This has a couple upsides:
    - If a `1.0.4` patch is released, you will still automatically make use of it,
    - You now know precisely which version is introducing the bug/problem (it is `1.1.0`, not `1.0.4` or something like that), even when new patches are put in place.

**Review by cdesaintguilhem — COMMENTED** [review]
As decided by @ycscaly, I'm pausing my review of this PR for now. There are still several core modules of the `pera-core` crate that I have _not_ reviewed, but I also made several comments in other places which might be helpful before we resume reviewing.

File: sdk/typescript/tsconfig.json (hunk 3 above)
Comment (zeevmoney) [reply sdk/typescript/tsconfig.json:9]: It's impossible to comment on this file.

File: crates/pera-analytics-indexer/Cargo.toml [hunk 4]
```diff
@@ -51,7 +51,7 @@ pera-json-rpc-types.workspace = true
 pera-package-resolver.workspace = true
 simulacrum.workspace = true
 arrow.workspace = true
-gcp-bigquery-client = "0.19.0"
+gcp-bigquery-client = "=0.18.0"
```
Comment (zeevmoney) [inline crates/pera-analytics-indexer/Cargo.toml:54]: reason for the downgrade?

File: crates/pera-core/src/execution_cache/cache_types.rs (hunk 1 above)
Comment (zeevmoney) [reply crates/pera-core/src/execution_cache/cache_types.rs:116]: No, there is no point to comment in places where code was removed.

**Comment by zeevmoney** [comment]
Merged in #351
